internal/uploader/uploader.go

```diff
@@ -0,0 +1,34 @@
+package uploader
+
+import (
+	"context"
+
+	"github.com/openshift/assisted-service/internal/common"
+	eventsapi "github.com/openshift/assisted-service/internal/events/api"
+	"github.com/openshift/assisted-service/pkg/k8sclient"
+	"github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+)
+
+//go:generate mockgen -source=uploader.go -package=uploader -destination=mock_uploader.go
+type Client interface {
+	UploadEvents(ctx context.Context, cluster *common.Cluster, eventsHandler eventsapi.Handler) error
+}
+
+type Config struct {
+	DataUploadEndpoint     string `envconfig:"DATA_UPLOAD_ENDPOINT" default:""`
+	AssistedServiceVersion string
+	EnableDataCollection   bool `envconfig:"ENABLE_DATA_COLLECTION" default:"true"`
```

Comment:
Do we need it also in the operator? If not, need to add this env var and set it to false in ASC [controller](https://github.com/openshift/assisted-service/blob/3c19e8e57e61279c145b3ea70e8fc63[…]nternal/controller/controllers/agentserviceconfig_controller.go](https://github.com/openshift/assisted-service/blob/3c19e8e57e61279c145b3ea70e8fc63e8f40d5bb/internal/controller/controllers/agentserviceconfig_controller.go#L1087)).
Also, need to disable it when running in a disconnected env(?) @masayag

Reply:
we do not need it in the operator but we do need it in the service for kube-api env. It should be propogated to it then.

Reply:
I think you're both correct, it needs to be in the operator since this feature is exposed for ACM/ZTP and it needs to be in the service for the kube-api (which I think it's already propagated [here](https://github.com/openshift/assisted-service/blob/3849ee79dcab20bf77873edfee96ee084f1beca1/cmd/main.go#L138), but correct me if I need to add additional changes to propagate it!).

For disconnected, it'll auto-fail when attempting to send the data, but is it better to set it to false directly? If we're setting to false directly, how do we know it's a disconnected environment?

Reply:
There are different build configuration depending on the environment (saas (staging and prod) and on prem). I would set it to false as default and set it intentionally to true on the selected environments. In this case you can deploy your feature gradually

Reply:
So do we need ENABLE_DATA_COLLECTION configurable also using the operator? If so, then it should be added to the [AgentServiceConfigSpec](https://github.com/openshift/assisted-service/blob/a3b5bacf62a593c3f1cb281654e01ffdd7d78a28/api/v1beta1/agentserviceconfig_types.go#L56) and propagate to the assisted-service [configmap](https://github.com/openshift/assisted-service/blob/3c19e8e57e61279c145b3ea70e8fc63e8f40d5bb/internal/controller/controllers/agentserviceconfig_controller.go#L1087) in agentserviceconfig_controller.
For deploying using podman, the configmap are [here](https://github.com/openshift/assisted-service/blob/84d7256be1c24f1ec4dc1dadf02bbc775d5ff3d3/deploy/podman/). But maybe having the default as `false` is indeed safer so we could explicitly enable it where we sure it's needed.

Reply:
I do not think it should be part of the AgentServiceConfigSpec. It belongs as other flags to the ai config map. The scrapping is done deep in the AI logic flow and has nothing to do with the operator

Reply:
Yeah, but the assisted-service configmap is getting overwritten on each asc reconcile loop. So the flag should be simply hardcoded in the cm if we don't want it in ASC.

Reply:
I think Daniel is correct, it needs to be configurable from the operator as well. I wasn't sure whether to 
1. Add the environment variable directly to the AgentServiceConfigSpec or
2. Just use the environment variable override annotation ([doc](https://github.com/openshift/assisted-service/blob/master/docs/operator.md#specifying-environmental-variables-via-configmap))

Which one do you all think is better?

Reply:
I did push the change to add these variables to the assisted-service [configmap](https://github.com/openshift/assisted-service/blob/3c19e8e57e61279c145b3ea70e8fc63e8f40d5bb/internal/controller/controllers/agentserviceconfig_controller.go#L1087) in agentserviceconfig_controller: https://github.com/openshift/assisted-service/blob/22b14d0e414ba9b529b14117f9b5609a2603c991/internal/controller/controllers/agentserviceconfig_controller.go#L1121-L1122

Reply:
> I think Daniel is correct, it needs to be configurable from the operator as well. I wasn't sure whether to
> 
> 1. Add the environment variable directly to the AgentServiceConfigSpec or
> 2. Just use the environment variable override annotation ([doc](https://github.com/openshift/assisted-service/blob/master/docs/operator.md#specifying-environmental-variables-via-configmap))
> 
> Which one do you all think is better?

Do we have any notion to how common this functionality could be for ACM/infrastructure-operator users?
It's just that using the override annotation is kind of a workaround, on the other hand, adding it to the ASC spec might be an overkill just for this. 
@filanov @avishayt  WDYT? Do we need `ENABLE_DATA_COLLECTION` configurable in the operator?

Reply:
How is ENABLE_DATA_COLLECTION differ from LOG_LEVEL for example?

Reply:
If it's enable until fail i don't see a reason not to enable by default in the operator.

Reply:
Shouldn't the default for `EnableDataCollection` here be set to `false` but in the config for the operator to `true`?

Reply:
I think we want data collection to be enabled for podman too, which isn't deployed via the operator, right? 

Then again, the variable could be set to true in the podman deploy's config file. 

Is it better to have it default to `false`?